version/version_test.go

```diff
@@ -39,3 +39,38 @@ func TestAsBigInt(t *testing.T) {
 		prev = next
 	}
 }
+
+func TestVersion_parseVersion(t *testing.T) {
+	require := require.New(t)
+
+	tests := map[string]struct {
+		major int
+		minor int
+		patch int
+		meta  string
+	}{
+		"v1.2.3":                       {major: 1, minor: 2, patch: 3},
+		"v1.2.3-alpha":                 {major: 1, minor: 2, patch: 3, meta: "alpha"},
+		"v1.2.3-alpha-dirty":           {major: 1, minor: 2, patch: 3, meta: "alpha-dirty"},
+		"some-non.stan-dard.12tag":     {},
+		"!`@#$%^&*()_{}|:<>?[]\\;',./": {},
+		"myTestTag":                    {},
+	}
+
+	originalGitTag := GitTag
+	for tag, want := range tests {
+		versionMajor = 0
+		versionMinor = 0
+		versionPatch = 0
+		versionMeta = ""
+
+		versionMajor, versionMinor, versionPatch, versionMeta = parseVersion(tag)
```

Comment:
Please do not use the variables for this test.

```suggestion
		versionMajor, versionMinor, versionPatch, versionMeta := parseVersion(tag)
```